File: Aula_Ricardo/Programação_orientada_a_objetos/go/Encapsulamento.go
package main

import "fmt"

type ContaBancaria struct {
	Titular string
	Saldo   float64
}

func (c *ContaBancaria) Depositar(adicionar float64) {
	c.Saldo += adicionar
}

func (c *ContaBancaria) Sacar(remover float64) {
	if remover > c.Saldo {
		c.Saldo = 0
	} else {
		c.Saldo -= remover
	}
}

func (c *ContaBancaria) ExibirSaldo() {
	fmt.Printf("Saldo: %.2f\n", c.Saldo)
}

func main() {
	titular1 := ContaBancaria{Titular: "etienne"}

	titular1.Depositar(100000)
	titular1.ExibirSaldo()
}

File: Aula_Ricardo/Programação_orientada_a_objetos/java/Encapsulamento.java
public class ContaBancaria {
    private String titular;
    private double saldo;

    public ContaBancaria(String titular) {
        this.titular = titular;
        this.saldo = 0.0;
    }

    public void depositar(double adicionar) {
        saldo += adicionar;
    }

    public void sacar(double remover) {
        saldo = Math.max(0, saldo - remover);
    }

    public void exibirSaldo() {
        System.out.println("Saldo: " + saldo);
    }

    public static void main(String[] args) {
        ContaBancaria titular1 = new ContaBancaria("etienne");
        titular1.depositar(100000);
        titular1.exibirSaldo();
    }
}
